src/components/forms/DebouncedCombobox.tsx

```diff
@@ -184,15 +185,30 @@ export function DebouncedCombobox(props: Props) {
 
           {records
             .filter((record) => !record.internal)
-            .map((record, index) => (
-              <Combobox.Option
-                className="cursor-pointer select-none relative py-2 px-3 text-gray-900 hover:bg-gray-100"
-                key={index}
-                value={record}
-              >
-                {record.label}
-              </Combobox.Option>
-            ))}
+            .map((record, index) => {
+              if (props.filterRecords) {
+                if (props.filterRecords(record, index))
+                  return (
+                    <Combobox.Option
+                      className="cursor-pointer select-none relative py-2 px-3 text-gray-900 hover:bg-gray-100"
+                      key={index}
+                      value={record}
+                    >
+                      {record.label}
+                    </Combobox.Option>
+                  );
+              } else {
+                return (
+                  <Combobox.Option
+                    className="cursor-pointer select-none relative py-2 px-3 text-gray-900 hover:bg-gray-100"
+                    key={index}
+                    value={record}
+                  >
+                    {record.label}
+                  </Combobox.Option>
+                );
+              }
+            })}
```

Comment:
This is a bad way of filtering stuff. If I get it correctly, you are performing filtering on each record?

I'd suggest the following way of doing this.

1. Passing filter function w/ callback to DebounceCombobox component that as arguments will return all arguments.
2. Trigger filter function on each change of the DebounceCombobox
3. Watch for change using useEffect and perform filtering top-level.